'Multiple elements for the same key' on sbt 1.3.0

When running sbt-whitesource on Akka with sbt 1.3.0-RC1 (https://github.com/akka/akka/pull/26935), I get:
```
[info] Processing com.typesafe.akka:akka-coordination:2.6-SNAPSHOT
[error] java.lang.RuntimeException: Unexpected error
[error] 	at sbtwhitesource.BaseAction.execute(WhiteSource.scala:72)
[error] 	at sbtwhitesource.WhiteSourcePlugin$.$anonfun$projectSettings$9(WhiteSourcePlugin.scala:152)
[error] 	at sbtwhitesource.WhiteSourcePlugin$.$anonfun$projectSettings$9$adapted(WhiteSourcePlugin.scala:149)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:270)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] 	at sbt.Execute.work(Execute.scala:279)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:270)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] Caused by: java.lang.RuntimeException: Multiple elements for the same key (com.github.jnr,jffi):
[error] 	ModuleInfo(com.github.jnr,jffi,1.2.15,Some((Artifact(jffi, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/com/github/jnr/jffi/1.2.15/jffi-1.2.15.jar), Map(artifactId -> jffi, project.parent.groupId -> org.sonatype.oss, project.parent.artifactId -> oss-parent, project.build.sourceEncoding -> UTF-8, sonatypeOssDistMgmtSnapshotsUrl -> https://oss.sonatype.org/content/repositories/snapshots/, project.version -> 1.2.15, pom.groupId -> com.github.jnr, maven.compiler.source -> 1.6, maven.compiler.target -> 1.6, maven.test.failure.ignore -> true, maven.test.skip -> true, project.parent.version -> 7, groupId -> com.github.jnr, parent.version -> 7, project.groupId -> com.github.jnr, version -> 1.2.15, pom.version -> 1.2.15, project.artifactId -> jffi, parent.groupId -> org.sonatype.oss, parent.artifactId -> oss-parent, make.exe -> make, pom.artifactId -> jffi), None),/home/travis/.cache/coursier/v1/https/repo1.maven.org/maven2/com/github/jnr/jffi/1.2.15/jffi-1.2.15.jar)))
[error] 	ModuleInfo(com.github.jnr,jffi,1.2.15,Some((Artifact(jffi, jar, jar, Some(native), Vector(), Some(https://repo1.maven.org/maven2/com/github/jnr/jffi/1.2.15/jffi-1.2.15-native.jar), Map(artifactId -> jffi, project.parent.groupId -> org.sonatype.oss, project.parent.artifactId -> oss-parent, project.build.sourceEncoding -> UTF-8, sonatypeOssDistMgmtSnapshotsUrl -> https://oss.sonatype.org/content/repositories/snapshots/, project.version -> 1.2.15, pom.groupId -> com.github.jnr, maven.compiler.source -> 1.6, maven.compiler.target -> 1.6, maven.test.failure.ignore -> true, maven.test.skip -> true, project.parent.version -> 7, groupId -> com.github.jnr, parent.version -> 7, project.groupId -> com.github.jnr, version -> 1.2.15, pom.version -> 1.2.15, project.artifactId -> jffi, parent.groupId -> org.sonatype.oss, parent.artifactId -> oss-parent, make.exe -> make, pom.artifactId -> jffi), None),/home/travis/.cache/coursier/v1/https/repo1.maven.org/maven2/com/github/jnr/jffi/1.2.15/jffi-1.2.15-native.jar)))
[error] 	at scala.sys.package$.error(package.scala:30)
[error] 	at sbtwhitesource.package$KeyByAndMergeSyntax$.$anonfun$keyByAndMerge$2(package.scala:26)
[error] 	at scala.Option.getOrElse(Option.scala:138)
[error] 	at sbtwhitesource.package$KeyByAndMergeSyntax$.$anonfun$keyByAndMerge$1(package.scala:26)
[error] 	at sbtwhitesource.package$KeyByAndMergeSyntax$.$anonfun$keyByAndMerge$1$adapted(package.scala:20)
[error] 	at scala.collection.Iterator.foreach(Iterator.scala:941)
[error] 	at scala.collection.Iterator.foreach$(Iterator.scala:941)
[error] 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
[error] 	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[error] 	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[error] 	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[error] 	at sbtwhitesource.package$KeyByAndMergeSyntax$.keyByAndMerge$extension(package.scala:20)
[error] 	at sbtwhitesource.BaseAction.moduleInfoByGA$1(WhiteSource.scala:162)
[error] 	at sbtwhitesource.BaseAction.moduleInfos$1(WhiteSource.scala:166)
[error] 	at sbtwhitesource.BaseAction.collectDependencyStructure(WhiteSource.scala:172)
[error] 	at sbtwhitesource.BaseAction.processProject(WhiteSource.scala:147)
[error] 	at sbtwhitesource.BaseAction.$anonfun$extractProjectInfos$2(WhiteSource.scala:83)
[error] 	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
[error] 	at scala.collection.Iterator.foreach(Iterator.scala:941)
[error] 	at scala.collection.Iterator.foreach$(Iterator.scala:941)
[error] 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
[error] 	at scala.collection.generic.Growable.$plus$plus$eq(Growable.scala:62)
[error] 	at scala.collection.generic.Growable.$plus$plus$eq$(Growable.scala:53)
[error] 	at scala.collection.immutable.VectorBuilder.$plus$plus$eq(Vector.scala:662)
[error] 	at scala.collection.immutable.VectorBuilder.$plus$plus$eq(Vector.scala:639)
[error] 	at scala.collection.TraversableOnce.to(TraversableOnce.scala:313)
[error] 	at scala.collection.TraversableOnce.to$(TraversableOnce.scala:311)
[error] 	at scala.collection.AbstractIterator.to(Iterator.scala:1429)
[error] 	at scala.collection.TraversableOnce.toVector(TraversableOnce.scala:309)
[error] 	at scala.collection.TraversableOnce.toVector$(TraversableOnce.scala:309)
[error] 	at scala.collection.AbstractIterator.toVector(Iterator.scala:1429)
[error] 	at sbtwhitesource.BaseAction.extractProjectInfos(WhiteSource.scala:83)
[error] 	at sbtwhitesource.CheckPoliciesAction.doExecute(WhiteSource.scala:361)
[error] 	at sbtwhitesource.BaseAction.execute(WhiteSource.scala:69)
[error] 	at sbtwhitesource.WhiteSourcePlugin$.$anonfun$projectSettings$9(WhiteSourcePlugin.scala:152)
[error] 	at sbtwhitesource.WhiteSourcePlugin$.$anonfun$projectSettings$9$adapted(WhiteSourcePlugin.scala:149)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:270)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] 	at sbt.Execute.work(Execute.scala:279)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:270)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] (whitesourceCheckPolicies) Unexpected error
```

These 2 ModuleInfo's seem to be the same, but contain different artifacts: the 'normal' and the 'native' version.

Note that this is not `sbt.librarymanagement.ModuleInfo` but `sbtwhitesource.BaseAction$ModuleInfo`.

--- a/project/build.properties
+++ b/project/build.properties
@@ -1,1 +1,1 @@
-sbt.version=1.2.6
+sbt.version=1.2.8

--- a/src/main/scala/sbtwhitesource/WhiteSource.scala
+++ b/src/main/scala/sbtwhitesource/WhiteSource.scala
@@ -238,7 +238,7 @@ sealed abstract class BaseAction(config: Config, childConfigs: Vector[ProjectCon
     def moduleInfoToTuple(m: ModuleInfo) = (m.groupId, m.artifactId, m.version)
 
     def artifactToTuple(a: Artifact) =
-      (a.name, a.extension, a.classifier, a.configurations, a.url, a.extraAttributes)
+      (a.name, a.extension, a.configurations, a.url, a.extraAttributes)
 
     if (moduleInfoToTuple(m1) != moduleInfoToTuple(m2)) None else {
       (m1.artifactAndJar, m2.artifactAndJar) match {
@@ -246,13 +246,21 @@ sealed abstract class BaseAction(config: Config, childConfigs: Vector[ProjectCon
         case (Some(_), None)                  => None
         case (None, None)                     => Some(m1)
         case (Some((a1, f1)), Some((a2, f2))) =>
-          val artifactAndJar = if (f1 != f2 || artifactToTuple(a1) != artifactToTuple(a2)) None else {
-            val t1 = a1.`type`
-            val t2 = a2.`type`
-            val finalType = if (t1 == t2) Some(t1)
-            else if (Set(t1, t2) == Set("jar", "bundle")) Some("bundle")
-            else None
-            finalType map (tpe => a1.withType(tpe) -> f1)
+          val artifactAndJar = if (artifactToTuple(a1) != artifactToTuple(a2)) None else {
+            val mergedType = Set(a1.`type`, a2.`type`).toSeq.sorted match {
+              case Seq(t1)              => Some(t1)
+              case Seq("bundle", "jar") => Some("bundle") // 'upgrade' "jar" to "bundle"
+              case _                    => None
+            }
+            val mergedClassifierAndFile = Set((a1.classifier, f1), (a2.classifier, f2)).toSeq.sorted match {
+              case Seq(x)                                  => Some(x)
+              case Seq(x @ (None, _), (Some("native"), _)) => Some(x) // discard "native" classifier
+              case _                                       => None
+            }
+            for {
+              tpe <- mergedType
+              (classifier, file) <- mergedClassifierAndFile
+            } yield a1.withType(tpe).withClassifier(classifier) -> file
           }
           artifactAndJar map (artifactAndJar => m1.copy(artifactAndJar = Some(artifactAndJar)))
       }

